pkg/platform/runtime/store/store.go

```diff
@@ -267,3 +271,11 @@ func (s *Store) StoreBuildPlan(build *bpModel.Build) error {
 	}
 	return nil
 }
+
+func (s *Store) BuildExpression() ([]byte, error) {
+	return fileutils.ReadFile(s.buildExpressionFile())
+}
```

Comment:
I think the build expression itself should be tied to the commit ID. So we can catch the commit and the buildexpression being out of sync. Technically that "shouldn't" happen, but at the moment if it does happen it will be difficult to diagnose.

Reply:
@Naatan What do you propose here? I don't see a way to do this without an internet connection (to query the build expression for the current commit ID). Note we already have https://github.com/ActiveState/cli/blob/mitchell/dx-1861/internal/runbits/buildscript/buildscript.go#L37-L41 to detect changes when a runtime mutation occurs. I don't know if that helps.

Reply:
I meant we should store the build expression in a way that it also embeds the commit ID, so we can assure at the time that we source this information that the two are in sync.

Eg. the build_expression file could have the commit ID as the first line, and the first validation we do is ensuring that this commit matches the commit that's in the local store.

Again this shouldn't really be an issue if we did our job right, but considering how new this code is and how hard it is to oversee all possible permutations it feels better to have some safety guards here since it'd always break in very vague ways.

Reply:
I committed something that ties the two together. Let me know if you expected something else or if it needs improvement.